indico/modules/events/contributions/operations.py

```diff
@@ -153,3 +155,30 @@ def delete_subcontribution(subcontrib):
     logger.info('Subcontribution %s deleted by %s', subcontrib, session.user)
     subcontrib.event_new.log(EventLogRealm.management, EventLogKind.negative, 'Subcontributions',
                              'Subcontribution "{}" has been deleted'.format(subcontrib.title), session.user)
+
+
+@no_autoflush
+def create_contribution_from_abstract(abstract, contrib_session=None):
+    event = abstract.event_new
+    contrib_person_links = {}
+    person_link_attrs = {'_title', 'address', 'affiliation', 'first_name', 'last_name', 'phone', 'author_type',
+                         'person', 'is_speaker'}
+    for abstract_person_link in abstract.person_links:
+        contrib_person_link_data = {}
+        for attr in person_link_attrs:
+            contrib_person_link_data[attr] = getattr(abstract_person_link, attr)
+        contrib_person_links[abstract_person_link.person.id] = ContributionPersonLink(**contrib_person_link_data)
+
+    duration = contrib_session.default_contribution_duration if contrib_session else timedelta(minutes=15)
+    custom_fields_data = {'custom_{}'.format(field_value.contribution_field.id): field_value.data for
+                          field_value in abstract.field_values}
+    create_contribution(event, {'friendly_id': abstract.friendly_id,
+                                'title': abstract.title,
+                                'duration': duration,
+                                'description': abstract.description,
+                                'abstract': abstract,
+                                'type': abstract.accepted_contrib_type,
+                                'track': abstract.accepted_track,
+                                'session': contrib_session,
+                                'person_link_data': {l: True for l in contrib_person_links.itervalues()}},
```

Comment:
lowercase-L should not be used as a variable name in python (just like uppercase-I). `link`?